nipype/workflows/fmri/fsl/preprocess.py

```diff
@@ -18,6 +18,25 @@ def getthreshop(thresh):
     return ['-thr %.10f -Tmin -bin' % (0.1 * val[1]) for val in thresh]
 
 
+
+def pickrun(files, whichrun):
+    """pick file from list of files"""
+
+    filemap = {'first': 0, 'last': -1, 'middle': len(files) // 2}
+
+    if isinstance(whichrun, str):
+        if whichrun not in filemap.keys():
+            raise(KeyError, 'Sorry, whichrun must be either integer index'
+                            'or string in form of "first", "last" or "middle')
+        else:
+            return files[filemap[whichrun]]
+
+    if isinstance(files, list):
```

Comment:
perhaps this if clause should be at the beginning to account for the potential that files is not a list, but a single filename.